Tryouts/Plugins/ApplicationPlugins/DwmSourceWpfApp/App.xaml.cs

```diff
@@ -0,0 +1,56 @@
+﻿/// ********************************************************************************************************
+///
+/// Morgan Stanley makes this available to you under the Apache License, Version 2.0 (the "License").
+/// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+/// See the NOTICE file distributed with this work for additional information regarding copyright ownership.
+/// Unless required by applicable law or agreed to in writing, software distributed under the License
+/// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and limitations under the License.
+/// 
+/// ********************************************************************************************************
+
+using MorganStanley.ComposeUI.Tryouts.Core.Abstractions;
+using MorganStanley.ComposeUI.Tryouts.Visuals.WindowsOnly.WinDwmSourceUtils;
+using NP.IoCy;
+using System;
+using System.Windows;
+
+namespace DwmImageOfWpfApp2
+{
+    /// <summary>
+    /// Interaction logic for App.xaml
+    /// </summary>
+    public partial class App : Application
+    {
+        private IoCContainer Container { get; } = new IoCContainer();
+
+        DwmVisual? _dwmVisual;
+
+        ISubscriptionClient _subscriptionClient;
+
+        public App()
+        {
+            Container.InjectPluginsFromSubFolders("Plugins/Services");
+            Container.CompleteConfiguration();
+            _subscriptionClient = Container.Resolve<ISubscriptionClient>();
+        }
+
+        protected override async void OnStartup(StartupEventArgs e)
+        {
+            base.OnStartup(e);
+
+            await _subscriptionClient.Connect(CommunicationsConstants.MachineName, CommunicationsConstants.Port);
```

Comment:
Does this mean that the application hosted here needs to 'connect' to something to be able to displayed as a DWM thumb?

Reply:
Connection is needed for communications between the shell and the WPF process behind the DWM thumb, not for the initial display. In this respect it is close to how it was implemented in MSDesktop.